src/ansys/mapdl/core/plotting/visualizer.py

```diff
@@ -0,0 +1,176 @@
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""Module for the MapdlPlotter class."""
+from ansys.tools.visualization_interface import Plotter
+from ansys.tools.visualization_interface.backends.pyvista import PyVistaBackendInterface
+from beartype.typing import Any, Iterable
+import pyvista as pv
+
+from ansys.mapdl.core.plotting.theme import MapdlTheme
+
+
+class MapdlPlotterBackend(PyVistaBackendInterface):
+    """Provides the plotter for PyMAPDL.
+
+    This class is an implementation of the ``PlotterInterface`` class from the ``ansys-visualizer`` package.
+    Picker is implemented in PyMAPDL-specific classes due to the characteristics of the library.
+
+    Parameters
+    ----------
+    use_trame : bool, optional
+        Whether to use the Trame interface or not. The default is ``False``.
+    theme : pv.DefaultTheme, optional
+        _description_, by default None1
+    """
+
+    def __init__(
+        self, use_trame: bool = False, plot_picked_names: bool = True, **plotter_kwargs
+    ):
+        """Initialize the ``MapdlPlotter`` class."""
+        super().__init__(
+            use_trame=use_trame, plot_picked_names=plot_picked_names, **plotter_kwargs
+        )
+
+    def plot_iter(
+        self,
+        plotting_list: Iterable[Any],
+        name_filter: str = None,
+        **plotting_options,
+    ) -> None:
+        """Add a list of objects to the plotter.
+
+        Parameters
+        ----------
+        plotting_list : Iterable[Any]
+            Iterable of objects to be added to the plotter.
+        name_filter : str, optional
+            Filter to be applied to the objects, by default None.
+        **plotting_options
+            Additional plotting options.
+        """
+        for plottable_object in plotting_list:
+            self.plot(plottable_object, name_filter, **plotting_options)
+
+    def plot(self, plottable_object: Any, name_filter: str = None, **plotting_options):
+        """Add an object to the plotter.
+
+        Parameters
+        ----------
+        plottable_object : Any
+            Object to be added to the plotter.
+        name_filter : str, optional
+            Filter to be applied to the object, by default None.
+        **plotting_options
+            Additional plotting options.
+        """
+        self.pv_interface.plot(
+            plottable_object=plottable_object,
+            name_filter=name_filter,
+        )
+
+
+class MapdlPlotter(Plotter):
+    """Plotter class for PyMapdl.
+
+    This class is an implementation of the PlotterInterface class from the ansys-visualizer package.
+    Picker is implemented in PyMAPDL specific classes due to the characteristics of the library.
+
+    Parameters
+    ----------
+    use_trame : bool, optional
+        Whether to use the trame interface or not, by default False.
```

Comment:
```suggestion
        Whether to use the Trame visualizer. The default is ``False``.
```
The brief description here is what is used in the Visualization Interface Tool.